### ISSUE TITLE
[Bug] Unity + Firebase Analytics iOS Deadlock+crash on start

### ISSUE THREAD

**Issue by Inspyro** [firebase/quickstart-unity#1156]
### [REQUIRED] Please fill in the following fields:

  * Unity editor version: 2019.3.18f1
  * Firebase Unity SDK version: 8.3.0
  * Source you installed the SDK: .unitypackage (.unitypackage or Unity Package Manager)
  * Problematic Firebase Component: Analytics(Auth, Database, etc.)
  * Other Firebase Components in use: None (Auth, Database, etc.)
  * Additional SDKs you are using: Facebook, Admob (Facebook, AdMob, etc.)
  * Platform you are using the Unity editor on: Windows (Mac, Windows, or Linux)
  * Platform you are targeting: iOS (iOS, Android, and/or desktop)
  * Scripting Runtime: IL2CPP(Mono, and/or IL2CPP)

### [REQUIRED] Please describe the issue here:
Add Firebase SDK to Unity, build and upload build to testflight
Download and install the build on iOS device
App immediately hangs (before unity splash screen is shown) on some devices (e.g. iPhone 7)
App crashes after around 10s of hangs

#### Steps to reproduce:
Repro rate: 100%

See attached crashlog from Testflight
[testflight_feedback.zip](https://github.com/firebase/quickstart-unity/files/7317691/testflight_feedback.zip)

**Comment by rusitschka** [comment]
Here are the pods in our project:

### Without Firebase
The app starts fine. Everything works.
```
-> Using FBAudienceNetwork (6.6.0)
-> Using FBSDKCoreKit_Basics (11.2.1)
-> Using Google-Mobile-Ads-SDK (8.12.0)
-> Using GoogleAppMeasurement (8.8.0)
-> Using GoogleMobileAdsMediationFacebook (6.6.0.0)
-> Using GoogleMobileAdsMediationUnity (3.7.5.0)
-> Using GoogleUserMessagingPlatform (2.0.0)
-> Using GoogleUtilities (7.5.2)
-> Using PromisesObjC (2.0.0)
-> Using UnityAds (3.7.5)
-> Using nanopb (2.30908.0)
```

### With Firebase
It hangs with deadlock right on application start.
```
-> Using FBAudienceNetwork (6.6.0)
-> Using FBSDKCoreKit_Basics (11.2.1)
-> Using Firebase (8.7.0)
-> Using FirebaseABTesting (8.8.0)
-> Using FirebaseAnalytics (8.7.0)
-> Using FirebaseCore (8.7.0)
-> Using FirebaseCoreDiagnostics (8.8.0)
-> Using FirebaseInstallations (8.8.0)
-> Using FirebaseRemoteConfig (8.7.0)
-> Using Google-Mobile-Ads-SDK (8.12.0)
-> Using GoogleAppMeasurement (8.7.0)
-> Using GoogleDataTransport (9.1.0)
-> Using GoogleMobileAdsMediationFacebook (6.6.0.0)
-> Using GoogleMobileAdsMediationUnity (3.7.5.0)
-> Using GoogleUserMessagingPlatform (2.0.0)
-> Using GoogleUtilities (7.5.2)
-> Using PromisesObjC (2.0.0)
-> Using UnityAds (3.7.5)
-> Using nanopb (2.30908.0)
```

### With Firebase and tweaked dependencies to 8.7.0
Still the same deadlock.
```
-> Using FBAudienceNetwork (6.6.0)
-> Using FBSDKCoreKit_Basics (11.2.1)
-> Using Firebase (8.7.0)
-> Using FirebaseABTesting (8.7.0)
-> Using FirebaseAnalytics (8.7.0)
-> Using FirebaseCore (8.7.0)
-> Using FirebaseCoreDiagnostics (8.7.0)
-> Using FirebaseInstallations (8.7.0)
-> Using FirebaseRemoteConfig (8.7.0)
-> Using Google-Mobile-Ads-SDK (8.7.0)
-> Using GoogleAppMeasurement (8.7.0)
-> Using GoogleDataTransport (9.1.0)
-> Using GoogleMobileAdsMediationFacebook (6.6.0.0)
-> Using GoogleMobileAdsMediationUnity (3.7.5.0)
-> Using GoogleUserMessagingPlatform (1.4.0)
-> Using GoogleUtilities (7.5.2)
-> Using PromisesObjC (2.0.0)
-> Using UnityAds (3.7.5)
-> Using nanopb (2.30908.0)
```

**Comment by eladleb** [comment]
I can confirm as well, after upgrading the firebase SDK, the game was stuck on:
+ (BOOL)isAppStoreReceiptSandbox (at GULAppEnvironmentUtil.m)

By adding FirebaseAppStoreReceiptURLCheckEnabled = NO in the game's plist, the game runs properly again.

I assume this can be easily reproduced with a test device that made TestFlight purchases.

![image](https://user-images.githubusercontent.com/1809338/139591860-9c7027ab-71f4-403d-adc6-a9d907b8aff7.png)

**Comment by Inspyro** [comment]
As other users already confirmed, when setting `FirebaseAppStoreReceiptURLCheckEnabled = NO` the issue no longer occurs. However I guess this should only be a temporary workaround, until the issue is fixed within Firebase?

**Comment by AlexeyPetruchikFortune** [comment]
+1 for this issue to be fixed properly. There was 4 releases (8.4.0, 8.5.0, 8.6.0, 8.6.1) after 8.3.0 and the issue is still not fixed.

**Comment by yingnierxiao** [comment]
8.6.0 crash,8.7 crash ,set FirebaseAppStoreReceiptURLCheckEnabled = false, no effect

**Comment by DFC005** [comment]
> +1 for this issue to be fixed properly. There was 4 releases (8.4.0, 8.5.0, 8.6.0, 8.6.1) after 8.3.0 and the issue is still not fixed.

And now 8.7.0 is out and it still doesn't fix the issue. Plenty of discussion in this issue as well: https://github.com/firebase/firebase-cpp-sdk/issues/700

**Comment by a-maurice** [comment]
Hi all,

We think we identified the problem, and have released a fix with 8.8.0.

For those curious, we think the issue was caused by changes with the newest version of xcode changing when the Objective C +load functions are called, which was resulting in the Firebase C++ SDK calling into some logging utility functions that were no longer safe to call.  We've changed it to no longer use these, and from our own testing we are now seeing that the underlying utility functions are being initialized before we call them now.

Sorry for the delay in getting a fix for this, the difficulty in reproducing this with a local build prevented us from identifying the root cause of the issue.
